### PR TITLE
[DONTMERGE] Various changes, refactoring into smaller PRs

### PR DESCRIPTION
**Pull request by Ho1yShif**
- Cleaned `meals_df` dataframe and added an explicit transpose
- Add `__str__` and `__repr__` methods to the class
- Separated functionality of google sheets authentication and JSON reading out of the `__init__` method into their own respective methods
- Corrected and added docstrings
- Additional cleanup including typos and spacing

### CODE REVIEW

File: plan_dinners.py [hunk 1]
```diff
@@ -43,41 +48,46 @@ def __init__(self):
...
+        self.start_of_week = today + datetime.timedelta(days=6-today.weekday())
+        self.week_timestamp = self.start_of_week.strftime('%b %d, %Y')
+
+        """Authenticate with Google Sheets API and read data from dinners.json"""
+        PlanDinners.setup_google_sheets_auth(self)
+        PlanDinners.read_dinners_json(self)
+
+    def __str__(self):
+        for day, meal in self.meal_schedule.items():
+            print(f"{day}: {meal.title()}")
```
Comment (boatbomber) [inline plan_dinners.py]: String method should build a string with these- it should not take any action of its own. It should return a string, and the user decides whether to print it, store it, etc.

File: plan_dinners.py [hunk 2]
```diff
@@ -43,41 +48,46 @@ def __init__(self):
...
+        return f"PlanDinners()"
+
+    def __repr__(self):
+        print(self.chosen_meals)
```
Comment (boatbomber) [inline plan_dinners.py]: Same critique here- it should not take action for the user.

**Review by boatbomber — CHANGES_REQUESTED** [review]
A few changes regarding the string representations, but the rest looks good!

File: plan_dinners.py (hunk 1 above)
Comment (Ho1yShif) [reply plan_dinners.py]: Updated the method to the following:
```python
def __str__(self):
        """Return a string representation of the weekly meal plan"""

        """Schedule meals if not already scheduled"""
        if not self.meals_scheduled_flag:
            PlanDinners.schedule_meals(self)
        return '\n'.join(f"{day}: {meal.title()}" for day, meal in self.meal_schedule.items())
```

File: plan_dinners.py (hunk 2 above)
Comment (Ho1yShif) [reply plan_dinners.py]: Updated the method to the following:
```python
def __repr__(self):
        """Return a string representation of the PlanDinners object"""
        if not self.meals_scheduled_flag:
            PlanDinners.schedule_meals(self)
        return f"PlanDinners(chosen_meals={self.chosen_meals})"
```

**Comment by Ho1yShif** [comment]
Fixed these issues! I'm excited to hear your thoughts :)

File: plan_dinners.py [hunk 3]
```diff
@@ -43,41 +48,52 @@ def __init__(self):
...
+
+        """Authenticate with Google Sheets API and read data from dinners.json"""
+        PlanDinners.setup_google_sheets_auth(self)
+        PlanDinners.read_dinners_json(self)
+
+    def __str__(self):
+        """Return a string representation of the weekly meal plan"""
+        """Schedule meals if not already scheduled"""
+        if not self.meals_scheduled_flag:
+            PlanDinners.schedule_meals(self)
```
Comment (boatbomber) [inline plan_dinners.py:65]: This is a side effect again! It should JUST return a string that shows an empty schedule. Getting a string to log somewhere should not update my schedule or google sheet!

**Review by boatbomber — CHANGES_REQUESTED** [review]

**Pull request closed without merging**